pkg/volume/fc/fc.go

```diff
@@ -126,17 +126,60 @@ func (plugin *fcPlugin) newMounterInternal(spec *volume.Spec, podUID types.UID,
 
 	return &fcDiskMounter{
 		fcDisk: &fcDisk{
-			podUID:  podUID,
+			podUID:  pod.UID,
+			volName: spec.Name(),
+			wwns:    fc.TargetWWNs,
+			lun:     lun,
+			wwids:   wwids,
+			manager: manager,
+			io:      &osIOHandler{},
+			plugin:  plugin},
+		fsType:     fc.FSType,
+		volumeMode: "",
+		readOnly:   readOnly,
+		mounter:    &mount.SafeFormatAndMount{Interface: mounter, Exec: exec},
+	}, nil
+}
+
+func (plugin *fcPlugin) NewBlockVolumeMapper(spec *volume.Spec, pod *v1.Pod, devicePath string, _ volume.VolumeOptions) (volume.BlockVolumeMapper, error) {
+	// Inject real implementations here, test through the internal function.
+	return plugin.newBlockVolumeMapperInternal(spec, pod, devicePath, &FCUtil{}, plugin.host.GetMounter(plugin.GetPluginName()), plugin.host.GetExec(plugin.GetPluginName()))
+}
+
+func (plugin *fcPlugin) newBlockVolumeMapperInternal(spec *volume.Spec, pod *v1.Pod, devicePath string, manager diskManager, mounter mount.Interface, exec mount.Exec) (volume.BlockVolumeMapper, error) {
+	// fc volumes used directly in a pod have a ReadOnly flag set by the pod author.
+	// fc volumes used as a PersistentVolume gets the ReadOnly flag indirectly through the persistent-claim volume used to mount the PV
+	fc, readOnly, err := getVolumeSource(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	var lun string
+	var wwids []string
+	if fc.Lun != nil && len(fc.TargetWWNs) != 0 {
```

Comment:
please extract a function that computes wwids, I see that NewMounter has the same code.

Reply:
Done.